dask/bag/core.py

```diff
@@ -482,19 +481,22 @@ def reduction(self, perpartition, aggregate, split_every=None,
         """
         if split_every is None:
             split_every = 8
-        a = next(names)
-        b = next(names)
+        token = tokenize(self, perpartition, aggregate, split_every)
+        a = 'reduction-part-' + token
```

Comment:
For user diagnostics it would be good to use a more informative name here, perhaps `dask.utils.funcname(perpartition)`